plugins/outputs/nats/nats.go

```diff
@@ -280,28 +319,66 @@ func (n *NATS) Write(metrics []telegraf.Metric) error {
 		return nil
 	}
 
+	// If the FieldName is included in the subject, we need to split the metric into multiple metrics
+	// each with a single field.
+	if n.includeFieldInSubject {
+		var newMetrics []telegraf.Metric
+		for _, m := range metrics {
+			newMetrics = append(newMetrics, splitMetricByField(m)...)
+		}
+		metrics = newMetrics
+	}
+
+	var bufSubject bytes.Buffer
+	var err error
+	var ack jetstream.PubAckFuture
+
+	subjectMetricMap := make(map[string][]telegraf.Metric)
+	for _, m := range metrics {
+		bufSubject.Reset()
+		err = n.tplSubject.Execute(&bufSubject, m.(telegraf.TemplateMetric))
+		if err != nil {
+			return fmt.Errorf("failed to execute subject template: %w", err)
+		}
+		subjectMetricMap[bufSubject.String()] = append(subjectMetricMap[bufSubject.String()], m)
+	}
+
 	var pafs []jetstream.PubAckFuture
 	if n.Jetstream != nil && n.Jetstream.AsyncPublish {
-		pafs = make([]jetstream.PubAckFuture, len(metrics))
+		pafs = make([]jetstream.PubAckFuture, 0, len(metrics))
 	}
 
-	for i, metric := range metrics {
-		buf, err := n.serializer.Serialize(metric)
-		if err != nil {
-			n.Log.Debugf("Could not serialize metric: %v", err)
+	for sub, metrics := range subjectMetricMap {
```

Comment:
Why can't you combine this with the `for` loop above and directly send the metric instead of intermediately storing the metrics?